### PR TITLE
[FTP-1778][PHP81] Fix PHP 8.1 types

### PR DESCRIPTION
**Pull request by costash**
[FTP-1778] Fix types for PHP 8

[FTP-1778]: https://getyourguide.atlassian.net/browse/FTP-1778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

https://sentry.io/organizations/getyourguide/issues/3881940091/?project=5719417&referrer=assigned_activity-email

### CODE REVIEW

File: src/containers/helper_traits/hacklib_immMapLike.php [hunk 1]
```diff
@@ -37,21 +37,25 @@ public function offsetGet($offset) {
...
     }
 
-    public function immutable() {
+    public function immutable()
+    {
       return $this;
     }
 
-    protected function hacklib_isImmutable() {
+    protected function hacklib_isImmutable()
```
Comment (pboos) [inline src/containers/helper_traits/hacklib_immMapLike.php:57]: No return type on this and `immutable()`? Same on the other `hacklib_imm` ones.

File: src/containers/helper_traits/hacklib_immVectorLike.php [hunk 2]
```diff
@@ -24,23 +24,27 @@ trait HACKLIB_ImmVectorLike {
     /**
      * identical to at, implemented for ArrayAccess
      */
-    public function offsetGet($offset) {
+    public function offsetGet($offset)
```
Comment (pboos) [inline src/containers/helper_traits/hacklib_immVectorLike.php:27]: Missing return type?

File: src/containers/helper_traits/hacklib_setLike.php [hunk 3]
```diff
@@ -26,15 +28,17 @@ public function offsetSet($offset, $value) {
...
+    {
       $this->remove($offset);
     }
 
     /**
      * Adds an element to this Set and returns itself. "$c->add($v)" is
      * equivalent to "$c[] = $v" (except that add() returns the Set).
      */
-    public function add($v) {
+    public function add($v)
```
Comment (pboos) [inline src/containers/helper_traits/hacklib_setLike.php:40]: Missing return type?

File: src/containers/helper_traits/hacklib_vectorLike.php [hunk 4]
```diff
@@ -24,26 +24,29 @@ trait HACKLIB_VectorLike {
     /**
      * identical to at, implemented for ArrayAccess
      */
-    public function &offsetGet($offset) {
+    public function &offsetGet($offset)
```
Comment (pboos) [inline src/containers/helper_traits/hacklib_vectorLike.php:27]: Missing return type?

**Review by pboos — APPROVED** [review]
Besides a few missing types (not sure if necessary), all is good.

File: src/containers/helper_traits/hacklib_immVectorLike.php (hunk 2 above)
Comment (costash) [reply src/containers/helper_traits/hacklib_immVectorLike.php:27]: This is mixed, which is a keyword added only in php 8. We can add attribute #ReturnTypeWillChange actually.

File: src/containers/helper_traits/hacklib_setLike.php (hunk 3 above)
Comment (costash) [reply src/containers/helper_traits/hacklib_setLike.php:40]: It's not defined in the interface. I won't add what's not strictly necessary for PHP 8.1

File: src/containers/helper_traits/hacklib_immMapLike.php (hunk 1 above)
Comment (costash) [reply src/containers/helper_traits/hacklib_immMapLike.php:58]: While we can add it, it's not strictly required for PHP 8.1. I wanna change as little as it's strictly required here. We're slowly deprecating thrift, so don't wanna spend too much effort here.

File: src/containers/helper_traits/hacklib_vectorLike.php (hunk 4 above)
Comment (costash) [reply src/containers/helper_traits/hacklib_vectorLike.php:28]: Added the `ReturnTypeWillChange` attribute.

**Pull request merged**